tendermint/src/vote/sign_vote.rs

```diff
@@ -0,0 +1,355 @@
+use crate::chain;
+use crate::Vote;
+use crate::{Error, Kind};
+use bytes::BufMut;
+use std::convert::TryFrom;
+use tendermint_proto::privval::SignedVoteResponse as RawSignedVoteResponse;
+use tendermint_proto::privval::{RemoteSignerError, SignVoteRequest as RawSignVoteRequest};
+use tendermint_proto::DomainType;
+use tendermint_proto::Error as DomainTypeError;
+
+/// SignVoteRequest is a request to sign a vote
+#[derive(Clone, PartialEq, Debug)]
+pub struct SignVoteRequest {
+    /// Vote
+    pub vote: Vote,
+    /// Chain ID
+    pub chain_id: chain::Id,
+}
+
+impl DomainType<RawSignVoteRequest> for SignVoteRequest {}
+
+impl TryFrom<RawSignVoteRequest> for SignVoteRequest {
+    type Error = Error;
+
+    fn try_from(value: RawSignVoteRequest) -> Result<Self, Self::Error> {
+        if value.vote.is_none() {
+            return Err(Kind::NoVoteFound.into());
+        }
+        Ok(SignVoteRequest {
+            vote: Vote::try_from(value.vote.unwrap())?,
+            chain_id: chain::Id::try_from(value.chain_id)?,
+        })
+    }
+}
+
+impl From<SignVoteRequest> for RawSignVoteRequest {
+    fn from(value: SignVoteRequest) -> Self {
+        RawSignVoteRequest {
+            vote: Some(value.vote.into()),
+            chain_id: value.chain_id.as_str().to_string(),
+        }
+    }
+}
+
+impl SignVoteRequest {
+    /// Create signable bytes from Vote.
+    pub fn to_signable_bytes<B>(&self, sign_bytes: &mut B) -> Result<bool, DomainTypeError>
+    where
+        B: BufMut,
+    {
+        self.vote
+            .to_signable_bytes(self.chain_id.clone(), sign_bytes)
+    }
+
+    /// Create signable vector from Vote.
+    pub fn to_signable_vec(&self) -> Result<Vec<u8>, DomainTypeError> {
+        self.vote.to_signable_vec(self.chain_id.clone())
+    }
+}
+
+/// SignedVoteResponse is a response containing a signed vote or an error
+#[derive(Clone, PartialEq)]
```

Comment:
```suggestion
#[derive(Clone, PartialEq, Debug)]
```

Reply:
Thanks, exactly the type of feedback I need to make the lib more usable. Added to the next commit.